quill-sql/src/main/scala/io/getquill/context/QueryExecution.scala

```diff
@@ -570,8 +570,9 @@ object PrepareDynamicExecution {
       matchingExternals: List[External],
       secondaryLifts: List[Planter[_, _, _]] = List()
   ): Either[String, (List[Planter[_, _, _]], List[Planter[_, _, _]])] = {
-    val encodeablesMap =
-      lifts.map(e => (e.uid, e)).toMap
+
+    var encodeablesMap =
+      lifts.groupBy(_.uid)
 
     val secondaryEncodeablesMap =
```

Comment:
We should also consider the `secondaryEncodeablesMap`, I don't know exactly where this is coming from, but we should apply the same fix there (I think) I might have some time to find reproducer for a bug that involve the `secondaryLifts` but for now I don't have one.

Reply:
Using the following test

```scala
  "reproduce with liftQuery" in {
    val repeated = Seq(Seq("a", "b"), Seq("c", "d"))

    val ctx = new SqlMirrorContext(MirrorSqlDialect, Literal)
    import ctx.*

    val query = dynamicQuery[Custom].filter(v =>
        repeated.map(l => 
          quote(liftQuery(l).contains(v.name))
        ).reduce((l,r) => quote(l || r))
    )
    val sql = ctx.translate(query)
    assert(sql == "SELECT v0.name FROM Custom v0 WHERE v0.name IN ('a', 'b') OR v0.name IN ('c', 'd')")
  }
```

The `secondaryLifts` parameters is always empty, so I'm not sure that it's related to `liftQuery` (at least not in this), so I guess we should not touch it if we don't understand what it's related to

Reply:
According to https://github.com/zio/zio-protoquill/commit/fb14ec1ba11afa2901df76c996311e8b079f1462#diff-0ed928e021d9d2075b190d1742422bf1f68d8a36d699741630f9fc4c0e1bb1d3R195
This is related to `lift` inside `liftQuery` and we should be able to reproduce/test with 
```
liftQuery(...).foreach(p => query[P].filter(pq => pq.id == lift(foo)).updateValue(p))
```

However I'm not sure that this is valid Quill code